drop-transfer/src/file/mod.rs

```diff
@@ -31,6 +31,7 @@ pub trait File {
     fn subpath(&self) -> &FileSubPath;
     fn size(&self) -> u64;
     fn mime_type(&self) -> &str;
+    fn direction(&self) -> TransferDirection;
```

Comment:
Using `self` here requires you to do a lookup of the file. This is not necessary. I suggest removing `self` from here or even better make it a `Transfer` trait method

Reply:
changed it to not take `Self`, making it part of the transfer trait would have been a little more tricky, since as far as i know, there is no way to partially specialize a trait

Reply:
I think i understood the initial comment incorrectly, fixed now :+1: